webdriver/tests/take_screenshot/user_prompts.py

```diff
@@ -2,63 +2,112 @@
+@pytest.mark.parametrize("dialog_type, retval", [
+    ("alert", None),
+    ("confirm", True),
+    ("prompt", ""),
+])
+def test_accept(check_user_prompt_closed_without_exception, dialog_type, retval):
+    check_user_prompt_closed_without_exception(dialog_type, retval)
 
 
 @pytest.mark.capabilities({"unhandledPromptBehavior": "accept and notify"})
-@pytest.mark.parametrize("dialog_type", ["alert", "confirm", "prompt"])
-def test_accept_and_notify(check_user_prompt_not_closed_without_exception, dialog_type):
-    check_user_prompt_not_closed_without_exception(dialog_type)
+@pytest.mark.parametrize("dialog_type, retval", [
+    ("alert", None),
+    ("confirm", True),
+    ("prompt", ""),
+])
+def test_accept_and_notify(check_user_prompt_closed_with_exception, dialog_type, retval):
+    check_user_prompt_closed_with_exception(dialog_type, retval)
 
 
 @pytest.mark.capabilities({"unhandledPromptBehavior": "dismiss"})
-@pytest.mark.parametrize("dialog_type", ["alert", "confirm", "prompt"])
-def test_dismiss(check_user_prompt_not_closed_without_exception, dialog_type):
-    check_user_prompt_not_closed_without_exception(dialog_type)
+@pytest.mark.parametrize("dialog_type, retval", [
+    ("alert", None),
+    ("confirm", False),
+    ("prompt", None),
+])
+def test_dismiss(check_user_prompt_closed_without_exception, dialog_type, retval):
+    check_user_prompt_closed_without_exception(dialog_type, retval)
 
 
 @pytest.mark.capabilities({"unhandledPromptBehavior": "dismiss and notify"})
-@pytest.mark.parametrize("dialog_type", ["alert", "confirm", "prompt"])
-def test_dismiss_and_notify(check_user_prompt_not_closed_without_exception, dialog_type):
-    check_user_prompt_not_closed_without_exception(dialog_type)
+@pytest.mark.parametrize("dialog_type, retval", [
+    ("alert", None),
+    ("confirm", False),
+    ("prompt", None),
+])
+def test_dismiss_and_notify(check_user_prompt_closed_with_exception, dialog_type, retval):
+    check_user_prompt_closed_with_exception(dialog_type, retval)
 
 
 @pytest.mark.capabilities({"unhandledPromptBehavior": "ignore"})
 @pytest.mark.parametrize("dialog_type", ["alert", "confirm", "prompt"])
-def test_ignore(check_user_prompt_not_closed_without_exception, dialog_type):
-    check_user_prompt_not_closed_without_exception(dialog_type)
+def test_ignore(check_user_prompt_not_closed_but_exception, dialog_type):
+    check_user_prompt_not_closed_but_exception(dialog_type)
 
 
-@pytest.mark.parametrize("dialog_type", ["alert", "confirm", "prompt"])
-def test_default(check_user_prompt_not_closed_without_exception, dialog_type):
-    check_user_prompt_not_closed_without_exception(dialog_type)
+@pytest.mark.parametrize("dialog_type, retval", [
+    ("alert", None),
+    ("confirm", False),
+    ("prompt", None),
+])
+def test_default(check_user_prompt_closed_with_exception, dialog_type, retval):
+    check_user_prompt_closed_with_exception(dialog_type, retval)
```

Comment:
nit: missing empty line at the end of the file.

Reply:
Put it back.